syntax = "proto3";
package etl_grpc.simplestore.bytes_store;
import "etl_grpc/basetypes/simplestore_error.proto";

message ReadRequest {
  string key = 1;
}

message ReadResponse {
  string key = 1;
  bytes bytes_content = 2;
  optional basetypes.simplestore_error.GrpcSimpleStoreError error = 3;
}

message WriteRequest {
  string key = 1;
  bytes bytes_content = 2;
}

message WriteResponse {
  string key = 1;
  optional basetypes.simplestore_error.GrpcSimpleStoreError error = 2;
}

service BytesSimpleStore {
  rpc Write(WriteRequest) returns (WriteResponse) {}
  rpc Read(ReadRequest) returns (ReadResponse) {}
}
